src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -216,18 +218,35 @@ public boolean isQuorumSynced(QuorumVerifier qv) {
     
     private final ServerSocket ss;
 
-    Leader(QuorumPeer self,LeaderZooKeeperServer zk) throws IOException {
+    Leader(QuorumPeer self,LeaderZooKeeperServer zk) throws IOException, X509Exception {
         this.self = self;
         try {
-            if (self.getQuorumListenOnAllIPs()) {
-                ss = new ServerSocket(self.getQuorumAddress().getPort());
+            if (self.shouldUsePortUnification()) {
```

Comment:
I think some level of refactoring that makes the constructor less clogged would be valuable here..